### PR TITLE
improved dockerfile builds

### PR DESCRIPTION
**Pull request by debris**
As noticed in #5435, our docker versions do not match tags. As described by @General-Beck this issue is currently solved by using custom build argument - `BUILD_TAG`. This pr simlifies and improves build process further.

Current build commands:

```bash
# branch that we want to build 
branch_or_tag_name="1.7"

# first clone of the repo
git clone https://github.com/paritytech/parity
cd parity
git checkout hub

# run docker build
# second clone of the repo occurs inside the container
docker build --build-arg BUILD_TAG=branch_or_tag_name --no-cache=true --tag ethcore/parity:branch_or_tag_name .
```

New build commands:

```bash
# branch that we want to build 
branch_or_tag_name="1.7"

# run docker build
# the only clone of the repo
git clone https://github.com/paritytech/parity
git checkout branch_or_tag_name

docker build -f docker/ubuntu/Dockerfile .
```

This pr also:
- makes every parity branch/tag buildable with stable/beta/nightly on centos/arm
- removes unneeded `ubuntu-dev` and `ubuntu-stable` dockerfiles

PS
*Needs to be backported, if merged

### CODE REVIEW

File: docker/centos/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,32 @@
 FROM centos:latest
 WORKDIR /build
+ADD . /build/parity
```
Comment (tomusdrw) [inline docker/centos/Dockerfile]: Add command should be moved right before `RUN cd parity` to utilized Docker cache for all deps and rust compilation.

**Comment by General-Beck** [comment]
We will also need to add readme and make corrections to the [wiki](https://github.com/paritytech/parity/wiki/Docker)

**Comment by 5chdn** [comment]
> We will also need to add readme and make corrections to the wiki

@General-Beck Can do it, what exactly changed for the user?

**Comment by gavofyork** [comment]
make sure this doesn't break the auto-updating integration please!

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Usage
+
+```docker build --no-cache=true -f docker/ubuntu/Dockerfile .```
```
Comment (asymmetric) [inline docker/README.md]: I would tag the image too by adding `-t parity`. This makes the output of `docker images` a bit more friendly:

```
❯ docker images
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
parity                         latest              9f8e2b065950        7 minutes ago       1.74 GB
```

Comment (asymmetric) [inline docker/README.md]: Also, why the `--no-cache` option?

Comment (debris) [inline docker/README.md]: I also don't know. `--no-cache` was here before (the file was moved from `docker/hub/README.md` -> `docker/README.md` and I simply kept it unchanged

Comment (General-Beck) [inline docker/README.md]: `--no-cache`can be removed

**Pull request merged**